mmdet3d/models/necks/dla_neck.py

```diff
@@ -0,0 +1,193 @@
+import math
+import numpy as np
+from mmcv.cnn import ConvModule, build_conv_layer
+from torch import nn as nn
+
+from mmdet.models.builder import NECKS
+
+
+class IDAUp(nn.Module):
+    """IDAUp module for upsamling different scale's features to same scale.
+
+    Args:
+        out_channel (int): Number of output channels for DeformConv.
+        in_channels_list (List[int]): List of input channels of multi-scale
+                                    feature map.
+        up_kernel_size_list (List[int]): List of size of the convolving
+                                        kernel of different scales.
+        norm_cfg (dict): Config dict for normalization layer. Default: None.
+        use_dcn (bool): If True, use DCNv2. Default: True.
+    """
+
+    def __init__(
+        self,
+        out_channel,
+        in_channels_list,
+        up_kernel_size_list,
+        norm_cfg=None,
+        use_dcn=True,
+    ):
+        super(IDAUp, self).__init__()
+        self.use_dcn = use_dcn
+        for i in range(1, len(in_channels_list)):
+            in_channel = in_channels_list[i]
+            up_kernel_size = int(up_kernel_size_list[i])
+            proj = ConvModule(
+                in_channel,
+                out_channel,
+                3,
+                padding=1,
+                bias=True,
+                conv_cfg=dict(type='DCNv2') if self.use_dcn else None,
+                norm_cfg=norm_cfg)
+            node = ConvModule(
+                out_channel,
+                out_channel,
+                3,
+                padding=1,
+                bias=True,
+                conv_cfg=dict(type='DCNv2') if self.use_dcn else None,
+                norm_cfg=norm_cfg)
+            up = build_conv_layer(
+                dict(type='deconv'),
+                out_channel,
+                out_channel,
+                up_kernel_size * 2,
+                stride=up_kernel_size,
+                padding=up_kernel_size // 2,
+                output_padding=0,
+                groups=out_channel,
+                bias=False)
+            setattr(self, 'proj_' + str(i), proj)
+            setattr(self, 'up_' + str(i), up)
+            setattr(self, 'node_' + str(i), node)
+
+    def forward(self, layers, start_level, end_level):
+        """Forward function.
+
+        Args:
+            layers (list[torch.Tensor]): features from multiple layers.
+            start_level (int): start layer for feature upsampling
+            end_level (int): end layer for feature upsampling
```

Comment:
First letter should be Capitalized. Please also refine other comments in the code